### ISSUE TITLE
Feature: Add custom events to be fired upon interaction

### ISSUE THREAD

**Issue by jmuheim** [NothingAG/adg-components#31]
Like in the original [Proof of Concept](https://josh.ch/dropdown/multi.html), we need some custom events to be fired upon interaction with the component.

🎬 https://user-images.githubusercontent.com/1814983/165909249-8b9a1dc3-17bc-40d2-b833-c65495300f67.gif

Specific events:

- When an option was selected/unselected (would also provide an argument that points to the respective option)
- When all options were unselected
- Maybe when the filter was changed (would also provide the current filter) → don't know of this is useful/necessary

See https://github.com/NothingAG/accessible-dropdown/blob/master/scripts.js#L69.

**Comment by wmaurer** [comment]
I have added the following events: `optionChanged`, `allOptionsUnselected`, `filterTermTextChanged`, `optionsDropdownOpened`, `optionsDropdownClosed`.

**Comment by jmuheim** [comment]
Cool thing, thank you, @wmaurer!

I updated the README accordingly:

![CleanShot 2022-05-02 at 12 43 39@2x](https://user-images.githubusercontent.com/1814983/166222258-ec75b7c1-080c-4cf8-91b9-1bdb287327ae.png)

Does this make sense?

I wonder whether we should rename `filterTermTextChanged` to `filterTermChanged` only? And is it good style / convention to use abbreviations like `prevFilterTermText` instead of `previousFilterTermText`?

**Comment by jmuheim** [comment]
I looked into the custom events. In general, they look good to me. Little question / idea:

For single select, why does changing a selection only fire an `optionChanged (selected: true)` for the newly selection option? Wouldn't it make sense to fire both a `optionChanged (selected: false)` for the previous selection and a `optionChanged (selected: true)` for the new one?

![CleanShot 2022-05-02 at 19 12 22@2x](https://user-images.githubusercontent.com/1814983/166293297-376b5b50-3b98-4941-910f-375213b451c3.png)

By the way: the `valueChanged` event seemed to be buggy. I removed it altogether, I think it's not relevant at the time being.

Thanks for your effort, @wmaurer! I'm closing this.
